spec/models/hets_instance_spec.rb

```diff
@@ -1,14 +1,52 @@
 require 'spec_helper'
 
 describe HetsInstance do
-  context 'when creating a hets instance' do
-    context 'and it has a reachable uri' do
-      let(:general_version) { '0.99' }
-      let(:specific_version) { '1409043198' }
+  let(:general_version) { Hets.minimum_version.to_s }
+  let(:specific_version) { Hets.minimum_revision.to_s }
+  let(:hets_instance) { create :local_hets_instance }
+
+  context 'when choosing a hets instance' do
+    context 'and there is no hets instance recorded' do
+      it 'should raise the appropriate error' do
+        expect { HetsInstance.choose }.
+          to raise_error(HetsInstance::NoRegisteredHetsInstanceError)
+      end
+    end
+
+    context 'and there is no acceptable hets instance' do
       let(:hets_instance) { create :local_hets_instance }
 
       before do
         stub_request(:get, "http://localhost:8000/version").
+          to_return(status: 500, body: "", headers: {})
+        hets_instance
+      end
+
+      it 'should raise the appropriate error' do
+        expect { HetsInstance.choose }.
+          to raise_error(HetsInstance::NoSelectableHetsInstanceError)
+      end
+    end
+
+    context 'and there is an acceptable hets instance' do
+      before do
+        stub_request(:get, "http://localhost:8000/version").
+          to_return(status: 200,
+                    body: "v#{general_version}, #{specific_version}",
+                    headers: {})
+        hets_instance
+      end
+
+      it 'should return that hets instance' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.